### PR TITLE
Added checking in Error function in Parser

### PR DESCRIPTION
**Pull request by Bamagik**
In cases of TokenStream not being defined. Like in issue#386 https://github.com/mozilla/rhino/issues/386

### CODE REVIEW

**Comment by gbrail** [comment]
Thanks! 

I'd love to see a test! 

I know that we don't have that many tests for the parser, but it seems that people are using Rhino to parse and help interpret JavaScript in lots of different projects and improving the parsers's quality via better code coverage would be a big help.

File: src/org/mozilla/javascript/NativeDate.java [hunk 1]
```diff
@@ -438,7 +438,6 @@ private static int YearFromTime(double t)
 
         double y = Math.floor(t / (msPerDay * 365.2425)) + 1970;
         double t2 = TimeFromYear(y);
-
```
Comment (szegedi) [inline src/org/mozilla/javascript/NativeDate.java]: This is irrelevant and should be removed.

Comment (szegedi) [inline src/org/mozilla/javascript/NativeDate.java]: This is irrelevant and should be removed.

File: src/org/mozilla/javascript/Parser.java [hunk 2]
```diff
@@ -233,8 +233,12 @@ void addError(String messageId, int position, int length) {
     }
 
     void addError(String messageId, String messageArg) {
-        addError(messageId, messageArg, ts.tokenBeg,
-                 ts.tokenEnd - ts.tokenBeg);
+        int beg = -1, end = -1;
```
Comment (szegedi) [inline src/org/mozilla/javascript/Parser.java]: I'd really like us to adopt a single-assignment style for local variables in Rhino whenever possible, so
```java
final int beg, end;
if (ts != null) {
    ...
} else {
    beg = end = -1;
}
```
would be much preferred.

**Comment by p-bakker** [comment]
The issue this PR fixes is already covered by #987, so closing this PR

**Pull request closed without merging**